internal/webhooks/runtime/extensionconfig_webhook.go

```diff
@@ -93,10 +101,121 @@ func (webhook *ExtensionConfig) validate(_ context.Context, _, _ *runtimev1.Exte
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 func (webhook *ExtensionConfig) ValidateDelete(_ context.Context, _ runtime.Object) error {
 	return nil
 }
+
+func validateExtensionConfigSpec(e *runtimev1.ExtensionConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	specPath := field.NewPath("spec")
+
+	if e.Spec.ClientConfig.URL == nil && e.Spec.ClientConfig.Service == nil {
+		allErrs = append(allErrs, field.Required(
+			specPath.Child("clientConfig"),
+			"either URL or Service must be defined",
+		))
+	}
+	if e.Spec.ClientConfig.URL != nil && e.Spec.ClientConfig.Service != nil {
+		allErrs = append(allErrs, field.Forbidden(
+			specPath.Child("clientConfig"),
+			"only one of URL or Service can be defined",
+		))
+	}
+
+	// Validate URL
+	if e.Spec.ClientConfig.URL != nil {
+		if uri, err := url.ParseRequestURI(*e.Spec.ClientConfig.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"must be a valid URL, e.g. https://example.com",
+			))
+		} else if uri.Scheme != "http" && uri.Scheme != "https" {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"'https' and 'http' are the only allowed URL schemes, e.g. https://example.com",
+			))
+		}
+	}
+
+	// Validate Service if defined
+	if e.Spec.ClientConfig.Service != nil {
+		// Validate that the name is not empty and is a Valid RFC1123 name.
+		if e.Spec.ClientConfig.Service.Name == "" {
+			allErrs = append(allErrs, field.Required(
+				specPath.Child("clientConfig", "service", "name"),
+				"must not be empty",
+			))
+		}
+
+		if errs := validation.IsDNS1035Label(e.Spec.ClientConfig.Service.Name); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "service", "name"),
+				e.Spec.ClientConfig.Service.Name,
+				"invalid name",
+			))
+		}
+
+		if e.Spec.ClientConfig.Service.Namespace == "" {
+			allErrs = append(allErrs, field.Required(
+				specPath.Child("clientConfig", "service", "namespace"),
+				"must not be empty",
+			))
+		}
+
+		if errs := validation.IsDNS1123Label(e.Spec.ClientConfig.Service.Namespace); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "service", "name"),
+				e.Spec.ClientConfig.Service.Name,
+				"invalid namespace",
+			))
+		}
+
+		if e.Spec.ClientConfig.Service.Path != nil {
+			path := *e.Spec.ClientConfig.Service.Path
+			if _, err := url.ParseRequestURI(path); err != nil {
+				allErrs = append(allErrs, field.Invalid(
+					specPath.Child("clientConfig", "service", "path"),
+					path,
+					"must be a valid URL path e.g. /path/to/hook",
+				))
+			}
+			if path[0:1] != "/" {
+				allErrs = append(allErrs, field.Invalid(
+					specPath.Child("clientConfig", "service", "path"),
+					path,
+					"must be a valid URL path e.g. /path/to/hook",
```

Comment:
Should we be slightly more explicit and write that the path has to start with /?

Reply:
I even removed this check because the above ParseRequestURI also handles that: https://go.dev/play/p/tjaxntZ0PAf

I added the err message to the above return value.